src/overlays/actors/ovl_Obj_Tokeidai/z_obj_tokeidai.h

```diff
@@ -3,14 +3,71 @@
 
 #include "global.h"
 
+#define OBJ_TOKEIDAI_TYPE(thisx) (((thisx)->params & 0xF000) >> 12)
+#define OBJ_TOKEIDAI_GET_CURRENT_HOUR(this) ((s32)((this)->currentTime * (24.0f / 0x10000)))
+#define OBJ_TOKEIDAI_GET_CURRENT_MINUTE(this) ((s32)((this)->currentTime * (360 * 2.0f / 0x10000)) % 30)
+#define OBJ_TOKEIDAI_GET_CLOCK_FACE_ROTATION(currentHour) ((s16)(currentHour * (0x10000 / 24.0f)))
+#define OBJ_TOKEIDAI_GET_OUTER_RING_OR_GEAR_ROTATION(currentMinute) ((s16)(currentMinute * (0x10000 * 12.0f / 360)))
+#define OBJ_TOKEIDAI_IS_TRANSFORMED() ((CURRENT_DAY == 3 && gSaveContext.time < CLOCK_TIME(6, 0)) || CURRENT_DAY >= 4)
+#define OBJ_TOKEIDAI_IS_STARTING_TRANSFORMATION_CS(globalCtx) \
+        (((globalCtx)->sceneNum == SCENE_CLOCKTOWER && gSaveContext.sceneSetupIndex == 2 && (globalCtx)->csCtx.unk_12 == 0) || \
+        ((globalCtx)->sceneNum == SCENE_00KEIKOKU && gSaveContext.sceneSetupIndex == 2 && (globalCtx)->csCtx.unk_12 == 0))
```

Comment:
The macros `OBJ_TOKEIDAI_IS_TRANSFORMED` and `OBJ_TOKEIDAI_IS_STARTING_TRANSFORMATION_CS` seems like it might have useful info that's better in the actual code instead of hidden behind a macro, but maybe that's just me. I don't have a particularly strong stance on that.

Also, what exactly does "Transformed" mean? It's mentioned in context of each individual part of this actor but at no point do I see mentioned what this transformation is?

Reply:
Starting with your second point, the Clock Tower normally looks like this:
![NKZE01_2021-11-20_09-11-40](https://user-images.githubusercontent.com/12245827/142735163-63d21e5a-700d-4143-ba28-8ebcc4d08e6a.png)
But in the final hours, it "transforms" to look like this:
![NKZE01_2021-11-20_09-12-30](https://user-images.githubusercontent.com/12245827/142735169-0c2dc11d-d205-4db1-b8fd-14c064f3ffdf.png)
I couldn't think of a better word for it other than "transformation". I can explain it in comments in the .c file, but I'd really like to nail down a good term for it first; any ideas?

As for the point about the macros, I pulled them out because they didn't provide any interesting local context; if you know what the macros mean (which, given your other point, is not a guarantee) then leaving the code there is just a bunch of "noise" that you have to parse.

Reply:
Perhaps "open", though people might confuse that with the actual steps that open up.

Reply:
Sorry for the delay on responding to this; I was focusing on finishing up EnFall. I took a look at the in-game text and the official Nintendo Power strategy guide, and pretty much it is universally described as "opening". I changed usages of "Transform" or "Transformed" to "Opening" or "Opened", and I added some text at the top of the file that better describes what this "opening" actually is for people who are unfamiliar with it. I'll leave this comment unresolved because it actually started with a discussion about macros, but I think the discussion about the name "Transformed" should be over.

Reply:
I agree with engineer, I'd prefer those to be in the code rather then hidden.

Reply:
Sounds good, I'll remove the macros and put it in code.